qiskit/pulse/library/waveform.py

```diff
@@ -41,9 +44,12 @@ def __init__(
                 If any sample's norm exceeds unity by less than or equal to epsilon
                 it will be clipped to unit norm. If the sample
                 norm is greater than 1+epsilon an error will be raised.
+            limit_amplitude: If True, then limit the amplitude of the waveform to 1.
         """
 
         samples = np.asarray(samples, dtype=np.complex_)
+        if limit_amplitude is not None:
+            self.limit_amplitude = limit_amplitude
```

Comment:
I would prefer to avoid having module-level variables. Is there a strong argument for why this is necessary?

Reply:
@taalexander Not sure I understand the remark.

The reason is that we cannot control the `limit_amplitude` argument to all the invocations of `Waveform`. With this PR one can create pulses with amplitude larger than 1 by 
```
Waveform.limit_amplitude = False

...
class Constant(qiskit.pulse.library.Constant):

    def validate_parameters(self) -> None:
        pass

....

pulse = Constant(100, 1.2, name='mypulse')

```
Somewhere in plotting or sending the pulse to a backend the `Waveform` is constructed (perhaps multiple times).

Without the class variable one would have to change the code at various places to get the right argument to `Waveform`.

Reply:
> @taalexander Not sure I understand the remark.

My apologies, the truncated Github view made me think `limit_amplitude` was a property of the module not the class.

> Somewhere in plotting or sending the pulse to a backend the Waveform is constructed (perhaps multiple times).

I see your point here. Ideally, I would prefer there to be a single location for program construction/compilation context to be stored, eg., a global `Context` class that a user may fetch/set through helper methods. However, this machinery is not in place currently so what you propose seems like a good solution.